Update docs before `0.19.0` release (#1275)
software-mansion/starknet.py: docs/guide/account_and_client.rst

Executing transactions
----------------------

To execute transactions on Starknet, use :meth:`~starknet_py.net.account.account.Account.execute_v1` or :meth:`~starknet_py.net.account.account.Account.execute_v3` methods from :ref:`Account` interface.
These methods will send :class:`~starknet_py.net.models.InvokeV1` and :class:`~starknet_py.net.models.InvokeV3` transactions respectively. To read about differences between transaction versions please visit `transaction types <https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions>`_ from the Starknet docs.

.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_executing_transactions.py
    :language: python
    :dedent: 4

Transaction Fee
---------------

> All methods within the :ref:`Account` that involve on-chain modifications require either to specify a maximum transaction fee or to use auto estimation.

Reviewer: Nit:
```suggestion
All methods within the :ref:`Account` that involve on-chain modifications require either specifying a maximum transaction fee or using auto estimation.
```

In the case of V1 and V2 transactions, the transaction fee, denoted in Wei, is configured by the ``max_fee`` parameter.
For V3 transactions, however, the fee is expressed in Fri and is determined by the ``l1_resource_bounds`` parameter.
To enable auto estimation, set the ``auto_estimate`` parameter to ``True``.

.. code-block:: python

    resp = await account.execute_v1(calls=call, auto_estimate=True)

.. warning::

>     It is strongly discouraged to use automatic fee estimation in production code as it may lead to unexpectedly high fee.

Reviewer: Nit:
```suggestion
    It is strongly discouraged to use automatic fee estimation in production code as it may lead to an unexpectedly high fee.
```

Executing transactions
----------------------

To execute transactions on Starknet, use :meth:`~starknet_py.net.account.account.Account.execute_v1` or :meth:`~starknet_py.net.account.account.Account.execute_v3` methods from :ref:`Account` interface.
These methods will send :class:`~starknet_py.net.models.InvokeV1` and :class:`~starknet_py.net.models.InvokeV3` transactions respectively. To read about differences between transaction versions please visit `transaction types <https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions>`_ from the Starknet docs.

.. codesnippet:: ../../starknet_py/tests/e2e/docs/guide/test_executing_transactions.py
    :language: python
    :dedent: 4

Transaction Fee
---------------

All methods within the :ref:`Account` that involve on-chain modifications require either to specify a maximum transaction fee or to use auto estimation.
In the case of V1 and V2 transactions, the transaction fee, denoted in Wei, is configured by the ``max_fee`` parameter.
For V3 transactions, however, the fee is expressed in Fri and is determined by the ``l1_resource_bounds`` parameter.

> To enable auto estimation, assign ``True`` to the ``auto_estimate`` parameter.

Reviewer 2: ```suggestion
To enable auto estimation, set the ``auto_estimate`` parameter to `True``.
```

.. code-block:: python

    resp = await account.execute_v1(calls=call, auto_estimate=True)

.. warning::

    Do not use automatic fee estimation in production code! It may lead to

>     very high fees paid as the amount returned by :meth:`~starknet_py.net.full_node_client.FullNodeClient.estimate_fee` may be arbitrarily large.

Reviewer 2: Let's rephrase this, to something like "it is discouraged to use auto estimation on prod"
if someone wants to use it, and is confident enough, let them.

> The returned estimated fee is multiplied by ``1.5`` for V1 and V2 transactions.

Reviewer 2: ```suggestion
The returned estimated fee is multiplied by ``1.5`` for V1 and V2 transactions to mitigate fluctuations in price.
```
maybe? I guess more info about this decision won't harm anyone

> For V3 transactions, ``max_amount`` and ``max_price_per_unit`` are scaled by ``1.1`` and ``1.5``.

Reviewer 2: ```suggestion
For V3 transactions, ``max_amount`` and ``max_price_per_unit`` are scaled by ``1.1`` and ``1.5`` respectively.
```
